### ISSUE TITLE
depgraph:graph throws StackOverflowError on parsing transitive dependencies

### ISSUE THREAD

**Issue by Eskibear** [ferstl/depgraph-maven-plugin#131]
For transitive dependencies, it goes into infinite loop and throws `StackOverflowError` exception. While `dependencies:tree` works well.

The command executed was:
```
mvn -N com.github.ferstl:depgraph-maven-plugin:graph -DgraphFormat=text -DshowDuplicates -DshowConflicts -DshowVersions -DshowGroupIds -DoutputDirectory=... -DoutputFileName=...
```
See https://github.com/microsoft/vscode-maven/issues/763#issuecomment-985572321

**Comment by ferstl** [comment]
Thanks for the investigation. I could reproduce the problem with a simple dependency to `arquillian-chameleon-junit-container-starter` and setting `-DshowConflicts`.

**Comment by ferstl** [comment]
The reason for the stack overflow is the conflicting dependency `rg.jboss.shrinkwrap.resolver:shrinkwrap-resolver-depchain` which forms a circle in the graph: 

```
com.github.ferstl:issue-131:1.0.0-SNAPSHOT:compile
\- org.arquillian.container:arquillian-chameleon-junit-container-starter:1.0.0.CR6:compile
   +- org.jboss.arquillian.junit:arquillian-junit-container:1.4.1.Final:compile
   |  +- org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-depchain:3.0.1:compile
   |  |  \- org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-impl-maven-embedded:3.0.1:runtime
   |  |     +- org.arquillian.spacelift:arquillian-spacelift:1.0.0.Alpha9:compile (omitted for conflict: 1.0.2)
   |  |     |  +- org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-depchain:3.0.1:compile (omitted for conflict: 2.1.0)

```

**Comment by ferstl** [comment]
I released version 3.3.1 of the plugin which fixes this issue. It should be available on Maven Central within the next few hours.
